File: python solution/fascinatingnumber.py
def isFascinating(n):
    strNum = str(n) + str(n*2) + str(n*3)
    freq = [0]*11
    
    for i in range( len(strNum) ):
        digit = int(strNum[i])
        freq[digit] += 1

    for i in range(1, 10) : 
        if freq[i] == 0 : 
            return False
    return True


num = int(input("Enter the 3 digit number: "))
if(isFascinating(num)):
    print(f'{num} is a fascinating number.')
else:
    print(f'{num} is a not fascinating number.')

File: java solution/fascinatingnumber.java
import java.util.*;

class Fascinating{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the 3 digit number: ");
        int num = sc.nextInt();
        if( isFascinating(num) ){
            System.out.println(num + " is a fascinating number.");
        }else{
            System.out.println(num + " is not a fascinating number.");
        }
    }

    public static boolean isFascinating(int n) {
        String strNum = ""+ n + (n*2) + (n*3);
        int freq[] = new int[11];

        for(int i = 0; i< strNum.length(); i++){
            int digit = strNum.charAt(i) - '0';
            freq[digit]++;
        }
        for(int i = 1; i<=9; i++){
            if(freq[i] != 0){
                return false;
            }
        }
        return true;
    }
}